src/server/rdb_test.cc

```diff
@@ -646,20 +646,11 @@ TEST_F(RdbTest, LoadHugeStream) {
   ASSERT_EQ(2000, CheckedInt({"xlen", "test:0"}));
 }
 
-TEST_F(RdbTest, SnapshotTooBig) {
-  // Run({"debug", "populate", "10000", "foo", "1000"});
-  //  usleep(5000);  // let the stats to sync
-  max_memory_limit = 100000;
-  used_mem_current = 1000000;
-  auto resp = Run({"debug", "reload"});
-  ASSERT_THAT(resp, ErrArg("Out of memory"));
-}
-
 TEST_F(RdbTest, LoadStream2) {
   auto ec = LoadRdb("RDB_TYPE_STREAM_LISTPACKS_2.rdb");
   ASSERT_FALSE(ec) << ec.message();
   auto res = Run({"XINFO", "STREAM", "mystream"});
-  EXPECT_THAT(
+  ASSERT_THAT(
```

Comment:
I am still not sure why this fails and I don't see how moving `SnapshotTooBig` below will have any effect or it will somehow stop reproducing since on each test case we `initialize everything from the start` so i don't see how changing `maxmemory` of one test case can affect another :shrug: 

Out of curiosity did you try running the test in a loop in a similar machine as the one it failed ?

Reply:
Good idea. But let's merge this fix and I will run the test in a loop.